### PR TITLE
changed: split VFP classes in base (ebos) and legacy

### PR DESCRIPTION
**Pull request by akva2**
to get rid of eigen usage in ebos based classes

### CODE REVIEW

**Comment by GitPaean** [comment]
What is the reason to split `VFPProperties` to be two types `VFPInjProp,VFPProdProp`, and also we are having them in template parameter for all the well model related class now?

**Comment by akva2** [comment]
- it was already split in two classes
- i am only dealing with the unstructured mess of a code which is already existing, with tentacles everywhere from everything.
- some objects holds the vfp and is shared between ebos and legacy. i have to template on this, because i cannot expose the additional functionality for legacy (usage of AutoDiffBlock / eigen) through any sort of virtual interface.

File: opm/autodiff/BlackoilWellModel.hpp [hunk 1]
```diff
@@ -70,7 +70,9 @@ END_PROPERTIES
 namespace Opm {
 
         /// Class for handling the blackoil well model.
-        template<typename TypeTag>
+        template<typename TypeTag,
+                 typename VFPInjProp=VFPInjProperties,
+                 typename VFPProdProp=VFPProdProperties>
```
Comment (atgeirr) [inline opm/autodiff/BlackoilWellModel.hpp]: This looks like something that will ripple through the whole code? Why must this be a template argument?

Comment (akva2) [reply opm/autodiff/BlackoilWellModel.hpp]: look at the code.

**Comment by GitPaean** [comment]
I think it is fine to make `VFPProperties` template if you want to avoid to make `VFPPropertiesLegacy` class (potential duplication of code). But the WellModel, WellInterface does not need to template of the `VFP` related types. 

It is clear, for the legacy well models, only Legacy related VFP things is needed, while for the `flow`, only the non-Legacy related VFP things is needed.  We will not have a `flow_something` needs different `VFP` related types. They really do not need to be template parameters.

**Comment by akva2** [comment]
fine. you come back with the non-templated solution.

**Comment by atgeirr** [comment]
> fine. you come back with the non-templated solution.

I'll try by creating a VFPPropertiesLegacy class.

**Comment by akva2** [comment]
and you will soon realize that those well classes have the VFP classes as members, and you have no virtual interface to work with so you WILL have to template on it.

**Pull request closed without merging**